custom/gcc_sangath/rules/custom_actions.py

```diff
@@ -0,0 +1,95 @@
+from corehq.apps.data_interfaces.models import (
+    AUTO_UPDATE_XMLNS,
+    CaseRuleActionResult,
+)
+from corehq.apps.hqcase.utils import update_case
+from corehq.form_processor.models import CommCareCase, CommCareCaseIndex
+from custom.gcc_sangath.const import (
+    DATE_OF_PEER_REVIEW_CASE_PROP,
+    MEAN_GENERAL_SKILLS_SCORE_CASE_PROP,
+    MEAN_TREATMENT_SPECIFIC_SCORE_CASE_PROP,
+    NEEDS_AGGREGATION_CASE_PROP,
+    NEEDS_AGGREGATION_NO_VALUE,
+    SESSION_CASE_TYPE,
+    SESSION_RATING_CASE_PROP,
+)
+
+
+def sanitize_session_peer_rating(session_case, rule):
+    """
+    For any session case
+    sanitize the peer_rating metrics based on child cases of type peer_rating
+    """
+    if session_case.type != SESSION_CASE_TYPE:
+        return CaseRuleActionResult()
+
+    peer_rating_cases = _get_peer_rating_cases(session_case)
```

Comment:
Is there guaranteed to be at least one peer rating case?

Reply:
I think so. If there are no peer rating cases
1. we won't do any update
2. I believe the app will set `needs_aggregation` only through peer_rating cases. So we should not get session cases with no peer_rating cases.
I can confirm with the app team.

Reply:
Got it - I misread the code earlier, I saw https://github.com/dimagi/commcare-hq/pull/31119/commits/28312a9f147519d07b6935037b5f3cf26452c145 remove an `if case_updates` check and missed that the `if peer_rating_cases` check is still there.

Reply:
cool, ya. As a setup i kept the possibility that there might not be case updates even if there are peer rating cases, but eventually it was that there will always be case updates if there are any peer rating cases.